shared/util/tf/tf_generator.go

```diff
@@ -0,0 +1,439 @@
+	datadogProviderBody.SetAttributeRaw("app_key", tokens("data.aws_ssm_parameter.dd_app_key.value"))
+	datadogProviderBody.SetAttributeRaw("api_key", tokens("data.aws_ssm_parameter.dd_api_key.value"))
+
+	_, err = tfFile.Write(hclFile.Bytes())
+
+	return err
+}
+
+func (tf TfGenerator) generateAwsProvider(rootBody *hclwrite.Body, alias, accountIdExpr, roleExpr string) error {
+	awsProviderBody := rootBody.AppendNewBlock("provider", []string{"aws"}).Body()
+	if alias != "" {
+		awsProviderBody.SetAttributeValue("alias", cty.StringVal(alias))
+	}
+	awsProviderBody.SetAttributeValue("region", cty.StringVal(*tf.happyConfig.AwsRegion()))
+
+	assumeRoleBlockBody := awsProviderBody.AppendNewBlock("assume_role", nil).Body()
+	assumeRoleBlockBody.SetAttributeRaw("role_arn", tokens(fmt.Sprintf("\"arn:aws:iam::%s:role/%s\"", accountIdExpr, roleExpr)))
+	awsProviderBody.SetAttributeRaw("allowed_account_ids", tokens(fmt.Sprintf("[\"%s\"]", accountIdExpr)))
+	return nil
+}
+
+func (tf *TfGenerator) GenerateVersions(srcDir string) error {
+	tfFile, err := os.Create(filepath.Join(srcDir, "versions.tf"))
+	if err != nil {
+		return errors.Wrap(err, "Unable to generate HCL code")
+	}
+	defer tfFile.Close()
+	hclFile := hclwrite.NewEmptyFile()
+
+	rootBody := hclFile.Body()
+	terraformBlockBody := rootBody.AppendNewBlock("terraform", nil).Body()
+	terraformBlockBody.SetAttributeValue("required_version", cty.StringVal(requiredTerraformVersion))
+	requiredProvidersBody := terraformBlockBody.AppendNewBlock("required_providers", nil).Body()
+
+	for _, provider := range requiredProviders {
+		p := cty.ObjectVal(map[string]cty.Value{
+			"source":  cty.StringVal(provider.Source),
+			"version": cty.StringVal(provider.Version),
+		})
+		requiredProvidersBody.SetAttributeValue(provider.Name, p)
+
+	}
+
+	_, err = tfFile.Write(hclFile.Bytes())
+
+	return err
+}
+
+func (tf *TfGenerator) GenerateOutputs(srcDir string, outputs []Output) error {
+	tfFile, err := os.Create(filepath.Join(srcDir, "outputs.tf"))
+	if err != nil {
+		return errors.Wrap(err, "Unable to generate HCL code")
+	}
+	defer tfFile.Close()
+	hclFile := hclwrite.NewEmptyFile()
+
+	rootBody := hclFile.Body()
+
+	sort.SliceStable(outputs, func(i, j int) bool {
+		return strings.Compare(outputs[i].Name, outputs[j].Name) < 0
+	})
+
+	for _, output := range outputs {
+		moduleOutputBody := rootBody.AppendNewBlock("output", []string{output.Name}).Body()
+		if len(output.Description) > 0 {
+			moduleOutputBody.SetAttributeValue("description", cty.StringVal(output.Description))
+		}
+		moduleOutputBody.SetAttributeValue("sensitive", cty.BoolVal(output.Sensitive))
+		tokens := hclwrite.TokensForTraversal(hcl.Traversal{
+			hcl.TraverseRoot{Name: "module"},
+			hcl.TraverseAttr{Name: "stack"},
+			hcl.TraverseAttr{Name: output.Name},
+		})
+		moduleOutputBody.SetAttributeRaw("value", tokens)
+	}
+
+	_, err = tfFile.Write(hclFile.Bytes())
+
+	return err
```

Comment:
static file, same comment as above

Reply:
This is not static. It pull module outputs and exposes them. If the module changes, outputs will change as well. For example, our current stack level outputs.tf is out of sync with the happy-env-* module outputs.